packages/calling-stateful-client/src/TogetherModeSubscriber.ts

```diff
@@ -2,49 +2,125 @@
 // Licensed under the MIT License.
 
 /* @conditional-compile-remove(together-mode) */
-import { TogetherModeCallFeature, TogetherModeVideoStream } from '@azure/communication-calling';
+import {
+  RemoteVideoStream,
+  TogetherModeCallFeature,
+  TogetherModeSeatingMap,
+  TogetherModeVideoStream
+} from '@azure/communication-calling';
 /* @conditional-compile-remove(together-mode) */
 import { CallContext } from './CallContext';
 /* @conditional-compile-remove(together-mode) */
 import { CallIdRef } from './CallIdRef';
-/**
- * @private
- */
-
+/* @conditional-compile-remove(together-mode) */
+import { InternalCallContext } from './InternalCallContext';
+/* @conditional-compile-remove(together-mode) */
+import { disposeCallFeatureView } from './CallFeatureStreamUtils';
+/* @conditional-compile-remove(together-mode) */
+import { convertSdkCallFeatureStreamToDeclarativeCallFeatureStream } from './Converter';
+/* @conditional-compile-remove(together-mode) */
+import { TogetherModeVideoStreamSubscriber } from './TogetherModeVideoStreamSubscriber';
+/* @conditional-compile-remove(together-mode) */
+import { CallFeatureStreamName } from './CallClientState';
 /* @conditional-compile-remove(together-mode) */
 /**
  * TogetherModeSubscriber is responsible for subscribing to together mode events and updating the call context accordingly.
  */
 export class TogetherModeSubscriber {
   private _callIdRef: CallIdRef;
   private _context: CallContext;
+  private _internalContext: InternalCallContext;
   private _togetherMode: TogetherModeCallFeature;
+  private _featureName: CallFeatureStreamName = 'togetherMode';
+  private _togetherModeVideoStreamSubscribers: Map<number, TogetherModeVideoStreamSubscriber>;
 
-  constructor(callIdRef: CallIdRef, context: CallContext, togetherMode: TogetherModeCallFeature) {
+  constructor(
+    callIdRef: CallIdRef,
+    context: CallContext,
+    internalContext: InternalCallContext,
+    togetherMode: TogetherModeCallFeature
+  ) {
     this._callIdRef = callIdRef;
     this._context = context;
+    this._internalContext = internalContext;
     this._togetherMode = togetherMode;
-
+    this._togetherModeVideoStreamSubscribers = new Map<number, TogetherModeVideoStreamSubscriber>();
     this.subscribe();
   }
 
   private subscribe = (): void => {
     this._togetherMode.on('togetherModeStreamsUpdated', this.onTogetherModeStreamUpdated);
+    this._togetherMode.on('togetherModeSceneUpdated', this.onSceneUpdated);
```

Comment:
What is scene updated vs seating updated? They both return `seatingInfo` - will scene updated ever contain information about the scene? (e.g. number of seats, image etc.)

Reply:
When together mode is active, participants can change the scene to a different one which will result in the seating positions of participants to be changed. which is why we return seatingInfo when this happens. Participants can also choose to reassign a seat to a different participant which is also why new seatingInfo is returned

Scene Updates only contain the coordinates of each participant in the together mode stream.

Reply:
Do we need to act on it or can we not subscribe to this? I assume seating updated also gets fired when the scene updates as the seating changes

Reply:
Yes we need to. Scene Change does not call the Seat update event from service side. Also there was request from the PM side to also have same thing on the SDK side

Reply:
So as a developer, if I listen to `togetherModeSeatingUpdated`, there are times when the seating updates but this is _not_ fired? That sounds like something we should fix?

Reply:
Yes when there's a scene change you will miss the newly seating information. The togetherModeSeatingUpdated comes during seat re-assignment

Reply:
(no changes to this PR for this thread given the SDK behavior)
Can we update the SDK to fire togetherModeSeatingUpdated in both scenarios? (Given both have updated the seating, the underlying reason should be opaque to the developer)